specification/2.0/appendix/performance-metrics/README.md

```diff
@@ -0,0 +1,194 @@
+It is possible to reference different meshes (nodes) using different scenes, this makes it possible to have the same texture assets but reference different primitives that use alternative materials (textures).  
+Ie one model could have 3 different scenes, where the same accessors (position, uv) are used but different primitives that reference materials with varying number of texture channels.  
+
+[nodeCount]  
+This represent the number of traversed nodes in a scene.  
+This is calculated by traversing nodes in each scene (depth or breadth first does not matter) - for each node increase the nodecount.  
+
+[vertexCount]  
+This represents the number of drawn vertices for each scene in the model.  
+This is calculated by traversing the nodes in each scene.  
+For each mesh use the POSITION attribute in each primitive, get the Accessor and add up the count field.  
+
+[primitiveCount]  
+This represents the number of primitives for each scene  
+Each primitive references a material with 0 or more texture sources, has attributes and accessors.  
+
+[textures]  
+This value represents the texture sources that are used by a scene.  
+This is the max value from the most 'complex' primitive that will be referenced by a Node in the scene.  
+The goal of this metric is to provide a worst case texture usage cost for rendering a pixel.  
+
+`textures`  
+1 BASECOLOR
+2 METALLICROUGHNESS  
+3 NORMAL  
+4 OCCLUSION  
+5 EMISSIVE  
+6 SPECULARGLOSS  
+7 DIFFUSE  
+8 CLEARCOAT  
+
+
+[materials]  
+This value represents the materials used by a scene.  
+This is the max value from the most 'complex' primitive that will be referenced by a Node in the scene.  
+The goal of this metric is to provide a worst case computational cost for rendering a pixel.  
+  
+1 PBR
+2 SPECULARGLOSS (KHR_materials_pbrSpecularGlossiness)
+2 UNLIT (KHR_materials_unlit) 
+
+
+[bounds]  
+This value represents the static (non animated) bounding volume for the scene.  
+It is calculated by traversing the nodes in each scene, applying node transform to primitive's MinMax values and updating the result.  
+The resulting MinMax will contain the total transformed volume the model occupies, ie concatenated MinMax of all primitives as they are transformed.  
+This will include translation, meaning that the bounds will not be centered.  
+
+
+[accessors]  
+This value represent the total vertex accessor usage of the model.  
+It is calculated by adding upp the accessors in the model, storing count, componentType and type.  
+This may give an indication of the runtime memory footprint of the buffers needed for the model.  
+componentType and type are taken from glTF specification for Accessor object.  
+`componentType`  
+5120 BYTE  
+5121 UNSIGNED_BYTE  
+5122 SHORT  
+5123 UNSIGNED_SHORT  
+5125 UNSIGNED_INT  
+5126 FLOAT  
+
+`type`  
+"SCALAR"  
+"VEC2"  
+"VEC3"  
+"VEC4"  
+"MAT2"  
+"MAT3"  
+"MAT4"  
+
+
+[textureSize]  
```

Comment:
If transmission size is important, there should be separate information about the size of the texture file. There can be a vast difference between the transmission (file) size and the GPU memory requirement, the latter being much larger when the former is .jpg or .png format.

What is the "texture array" referenced here? If it refers to an array texture, there is not much difference between iterating over the images in one of those and iterating over a set of mipmaps to calculate size.

For the purpose of calculating GPU size (& file size) most block-ccompressed textures have 1 "component", a small number of formats have 2. You will need to provide the size of those components (that is the block size) and the dimension (num_pixels_x * num_pixels_y * num_pixels z) of the block for a  size calculation to be possible.

For uncompressed formats you will need to provide the component type just as you are for accessors. Maybe today glTF only uses 8-bit components but I'm sure that will change.

Reply:
Sure, that's why there is a texture array - this is the textures (images) that are defined in the asset together with the sizes so that the runtime uncompressed memory requirements can be calculated for textures.
This will currently not handle compressed textures or any other component types than are currently supported (8 bit)